### PR TITLE
Setup Wizard: Configure Validators for fields

### PR DESCRIPTION
**Pull request by Abhinegi2**
closes:  #2118

### Visible/Frontend Changes
- [x] Added New fields in Entity fields Advance option to configure validations
- [x] Updated a Save functionality it will also save the validator property of entity type.

### CODE REVIEW

File: src/app/core/admin/admin-entity-details/admin-entity-field/admin-entity-field.component.html [hunk 1]
```diff
@@ -145,7 +145,7 @@ <h2 mat-dialog-title i18n>Configure Field "{{ entitySchemaField.label }}"</h2>
       <mat-tab
         label="Advanced Options & Validation [COMING SOON]"
         i18n-label
-        [disabled]="true"
+        [disabled]="false"
```
Comment (sleidig) [inline src/app/core/admin/admin-entity-details/admin-entity-field/admin-entity-field.component.html:148]:
```suggestion
```

File: src/app/core/admin/admin-entity-details/admin-entity-field/admin-entity-field.component.html [hunk 2]
```diff
@@ -145,7 +145,7 @@ <h2 mat-dialog-title i18n>Configure Field "{{ entitySchemaField.label }}"</h2>
       <mat-tab
         label="Advanced Options & Validation [COMING SOON]"
```
Comment (sleidig) [inline src/app/core/admin/admin-entity-details/admin-entity-field/admin-entity-field.component.html:146]:
```suggestion
        label="Advanced Options & Validation"
```

File: src/app/core/admin/admin-entity-details/admin-entity-field/configure-entity-field-validator/configure-entity-field-validator.component.scss [hunk 3]
```diff
@@ -0,0 +1,9 @@
+.validator-form-cell {
+    display: flex;
+    flex-direction: column;
+    // set the width of each form field to 100% in every form component that is a descendent
+    // of the columns-wrapper class
+    mat-form-field {
+      width: 80%;
+    }
+}
```
Comment (sleidig) [inline src/app/core/admin/admin-entity-details/admin-entity-field/configure-entity-field-validator/configure-entity-field-validator.component.scss:9]: for a flex layout, you can use our global class "flex-column" without having to define your own css

File: src/app/core/admin/admin-entity-details/admin-entity-field/configure-entity-field-validator/configure-entity-field-validator.component.ts [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+import { EntityConstructor } from "../../../../entity/model/entity";
+import { MatCheckboxModule } from "@angular/material/checkbox";
+import { EntitySchemaField } from "app/core/entity/schema/entity-schema-field";
+import { DynamicValidator } from "app/core/common-components/entity-form/dynamic-form-validators/form-validator-config";
+
+@Component({
+  selector: "app-configure-entity-field-validator",
+  standalone: true,
+  imports: [
+    NgForOf,
```
Comment (sleidig) [inline src/app/core/admin/admin-entity-details/admin-entity-field/configure-entity-field-validator/configure-entity-field-validator.component.ts:24]: I think ngFor is not used (anymore) in this component?

File: src/app/core/admin/admin-entity-details/admin-entity-field/configure-entity-field-validator/configure-entity-field-validator.component.ts [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+    MatCheckboxModule,
+    FontAwesomeModule,
+    ReactiveFormsModule,
+    MatButtonModule,
+  ],
+  templateUrl: "./configure-entity-field-validator.component.html",
+  styleUrl: "./configure-entity-field-validator.component.scss",
+})
+export class ConfigureValidatorPopupComponent {
+  fieldId: string;
```
Comment (sleidig) [inline src/app/core/admin/admin-entity-details/admin-entity-field/configure-entity-field-validator/configure-entity-field-validator.component.ts:38]: not used? Can you go through the component and check whether all the inputs and properties are actually required for you final implementation?

File: src/app/core/admin/admin-entity-details/admin-entity-field/configure-entity-field-validator/configure-entity-field-validator.component.html [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+    <mat-form-field>
+      <mat-label i18n>Maximum Value</mat-label>
+      <input formControlName="max" matInput type="number" />
+    </mat-form-field>
+
+    <mat-form-field>
+      <mat-label i18n>Regex Pattern</mat-label>
+      <input formControlName="regex" matInput />
+    </mat-form-field>
+    <mat-checkbox i18n formControlName="required"> Is Required? </mat-checkbox>
```
Comment (sleidig) [inline src/app/core/admin/admin-entity-details/admin-entity-field/configure-entity-field-validator/configure-entity-field-validator.component.html:17]: these checkboxes do not display as a new line yet:
image

maybe needs to be wrapped in a form-field or div or something?

File: src/app/core/admin/admin-entity-details/admin-entity-field/configure-entity-field-validator/configure-entity-field-validator.component.html [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+  <mat-tab-group formGroupName="validatorForm">
+    <mat-form-field>
+      <mat-label i18n>Minimum Value</mat-label>
+      <input formControlName="min" matInput type="number" />
+    </mat-form-field>
+
+    <mat-form-field>
+      <mat-label i18n>Maximum Value</mat-label>
+      <input formControlName="max" matInput type="number" />
+    </mat-form-field>
```
Comment (sleidig) [inline src/app/core/admin/admin-entity-details/admin-entity-field/configure-entity-field-validator/configure-entity-field-validator.component.html:11]: hide if entityField dataType is not "number"

File: src/app/core/admin/admin-entity-details/admin-entity-field/configure-entity-field-validator/configure-entity-field-validator.component.html [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+
+    <mat-form-field>
+      <mat-label i18n>Regex Pattern</mat-label>
+      <input formControlName="regex" matInput />
+    </mat-form-field>
```
Comment (sleidig) [inline src/app/core/admin/admin-entity-details/admin-entity-field/configure-entity-field-validator/configure-entity-field-validator.component.html:16]: hide regex and "valid email" if datatype is not "string"

**Review by sleidig — CHANGES_REQUESTED** [review]

**Comment by Abhinegi2** [comment]
@sleidig did requested changes can you please check now?

File: src/app/core/admin/admin-entity-details/admin-entity-field/configure-entity-field-validator/configure-entity-field-validator.component.html [hunk 9]
```diff
@@ -0,0 +1,33 @@
+<form class="validator-form-cell flex-column" [formGroup]="form">
+  <mat-tab-group formGroupName="validatorForm">
```
Comment (sleidig) [inline src/app/core/admin/admin-entity-details/admin-entity-field/configure-entity-field-validator/configure-entity-field-validator.component.html:2]: my IDE remarks that mat-tab-group is not allowed here (must be within the template that has the mat-tab --> take out of the template here and wrap in admin-entity-field instead)

**Comment by Abhinegi2** [comment]
@sleidig did the requested changes, please have a look now.

**Review by sleidig — APPROVED** [review]

File: src/app/core/admin/admin-entity-details/admin-entity-field/configure-entity-field-validator/configure-entity-field-validator.component.spec.ts [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+import { MatCheckboxModule } from "@angular/material/checkbox";
+import { MatFormFieldModule } from "@angular/material/form-field";
+import { MatInputModule } from "@angular/material/input";
+import { MatButtonModule } from "@angular/material/button";
+import { FontAwesomeModule } from "@fortawesome/angular-fontawesome";
+import { ConfigureEntityFieldValidatorComponent } from "./configure-entity-field-validator.component";
+import { BrowserAnimationsModule } from "@angular/platform-browser/animations";
+import { FontAwesomeTestingModule } from "@fortawesome/angular-fontawesome/testing";
+
+fdescribe("ConfigureEntityFieldValidatorComponent", () => {
```
Comment (sleidig) [inline src/app/core/admin/admin-entity-details/admin-entity-field/configure-entity-field-validator/configure-entity-field-validator.component.spec.ts:12]:
```suggestion
describe("ConfigureEntityFieldValidatorComponent", () => {
```

**Pull request merged**